### PR TITLE
Added pending tx observable, querying iroha, updated gitignore

### PR DESCRIPTION
**Pull request by Mingela**

### CODE REVIEW

File: src/main/java/iroha/validation/config/ValidationServiceContext.java [hunk 1]
```diff
@@ -29,7 +28,6 @@ public ValidationServiceContext(
     this.validators = validators;
     this.transactionProvider = transactionProvider;
     this.transactionSigner = transactionSigner;
-    this.keyPair = keyPair;
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/config/ValidationServiceContext.java]: I guess `keyPair` is not used anymore.

File: src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java [hunk 2]
```diff
@@ -3,15 +3,28 @@
 import iroha.protocol.TransactionOuterClass.Transaction;
 import iroha.validation.transactions.signatory.TransactionSigner;
 import java.security.KeyPair;
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+import jp.co.soramitsu.iroha.java.IrohaAPI;
 
 public class TransactionSignerImpl implements TransactionSigner {
 
+  private IrohaAPI irohaAPI;
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java]: Can be `final`

File: src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java [hunk 3]
```diff
@@ -3,15 +3,28 @@
...
+  private KeyPair keyPair;
+
+  public TransactionSignerImpl(String host, int port, KeyPair keyPair) {
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java]: `host` and `keyPair` can be null

File: src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import jp.co.soramitsu.iroha.java.BlocksQueryBuilder;
+import jp.co.soramitsu.iroha.java.IrohaAPI;
+import jp.co.soramitsu.iroha.java.Query;
+import jp.co.soramitsu.iroha.java.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BasicTransactionProvider implements TransactionProvider {
+
+  private static final Logger logger = LoggerFactory.getLogger(MQTransactionProvider.class);
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java]: Why `getLogger(MQTransactionProvider.class)`? Is it correct? Do we use `MQTransactionProvider.class` on puprose?

File: src/main/java/iroha/validation/config/ValidationServiceContext.java (hunk 1 above)
Comment (Mingela) [reply src/main/java/iroha/validation/config/ValidationServiceContext.java]: Thanks! Done

File: src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java (hunk 2 above)
Comment (Mingela) [reply src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java]: Thanks! Done

File: src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java (hunk 3 above)
Comment (Mingela) [reply src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java]: Thanks! Done

File: src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java (hunk 4 above)
Comment (Mingela) [reply src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java]: Thanks! Done

File: src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    Objects.requireNonNull(irohaAPI, "Iroha API must not be null");
+    if (Strings.isNullOrEmpty(accountId)) {
+      throw new IllegalArgumentException("Account ID must not be neither null or empty");
+    }
+    Objects.requireNonNull(keyPair, "Keypair must not be null");
+
+    this.irohaAPI = irohaAPI;
+    this.accountId = accountId;
+    this.keyPair = keyPair;
+    Executors.newScheduledThreadPool(1)
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java]: Running threads from constructors can be risky. In this particular case, I cannot see any problems. But there may be problems in the future. Please take a brief look at [this article](https://www.ibm.com/developerworks/java/library/j-jtp0618/index.html?ca=drs-).

Comment (Mingela) [reply src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java]: Done

**Pull request merged**